src/firmware/shared/circular_buffer.h

```diff
@@ -0,0 +1,42 @@
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "shared/constants.h"
+
+/**
+ * This struct represents the circular buffer properties
+ */
+typedef struct CircularBuffer CircularBuffer_t;
+
+/**
+ * Initialize the circular_buffer struct members
+ * @param  cbuffer The circular_buffer
+ * @param  size Size of the circular_buffer to set
+ */
+CircularBuffer_t *circular_buffer_create(CircularBuffer_t *cbuffer, int size);
```

Comment:
As a general note, anything that denotes a size of an array or an index should probably be a `size_t` instead of an `int` (for context: https://stackoverflow.com/questions/2550774/what-is-size-t-in-c)

Reply:
Thanks for the link! Fixed to use `size_t`.